src/test/java/com/google/sps/servlets/EventServletTest.java

```diff
@@ -0,0 +1,142 @@
+    when(mockRequest.getParameter("search")).thenReturn(search.getText());
+    when(mockRequest.getParameter("category")).thenReturn(search.getCategory());
+    when(mockRequest.getParameter("start")).thenReturn(search.getStart());
+    when(mockRequest.getParameter("end")).thenReturn(search.getEnd());
+    when(mockRequest.getParameter("duration")).thenReturn(search.getDuration());
+    when(mockRequest.getParameter("location")).thenReturn(search.getLocation());
+  }
+
+  private void setWriterAndCreateEvent() throws IOException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    new EventServlet(mockUserStorage, mockEventStorage).doGet(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithNullPath_DoesSearch() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn(null);
+
+    setWriterAndCreateEvent();
+
+    verify(mockEventStorage, atLeast(1)).getSearchedEvents(any(Search.class));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithEmptyPath_DoesSearch() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn("");
+
+    setWriterAndCreateEvent();
+
+    verify(mockEventStorage, atLeast(1)).getSearchedEvents(any(Search.class));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithSlashPath_DoesSearch() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn("/");
+
+    setWriterAndCreateEvent();
+
+    verify(mockEventStorage, atLeast(1)).getSearchedEvents(any(Search.class));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithDifferentNonEmptyPath_DoesntSearch() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn("/gcalendar");
+
+    new EventServlet(mockUserStorage, mockEventStorage).doGet(mockRequest, mockResponse);
+
+    verify(mockEventStorage, never()).getSearchedEvents(any(Search.class));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithNullPathAndGivenParameters_DoesSearch() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn(null);
+
+    Search search = new Search("", "all", "2020-01-01", "2020-12-31", "60", "ChIJGaK-SZcLkEcRA9wf5_GNbuY");
+    
+    setParameters(search);
+
+    setWriterAndCreateEvent();
+    
+    verify(mockEventStorage, atLeast(1)).getSearchedEvents(eq(search));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithEmptyPathAndGivenParameters_DoesSearch() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn("");
+
+    Search search = new Search("", "all", "2020-01-01", "2020-12-31", "60", "ChIJGaK-SZcLkEcRA9wf5_GNbuY");
+    
+    setParameters(search);
+
+    setWriterAndCreateEvent();
+    
+    verify(mockEventStorage, atLeast(1)).getSearchedEvents(eq(search));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithSlashPathAndGivenParameters_DoesSearch() throws IOException {
```

Comment:
Please also add a test that the servlet writes the output correctly if something is returned in the `getSearchedEvents(...)` call.